.github/workflows/model_uploader.yml

```diff
@@ -0,0 +1,321 @@
+
+  # Step 4: Ask for manual approval from the CODEOWNERS
+  manual-approval:
+    needs: model-auto-tracing
+    runs-on: 'ubuntu-latest'
+    permissions:
+      issues: write
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      - name: Get Approvers
+        id: get_approvers
+        run: |
+          echo "approvers=$(cat .github/CODEOWNERS | grep @ | tr -d '* ' | sed 's/@/,/g' | sed 's/,//1')" >> $GITHUB_OUTPUT
+      - name: Create Issue Body
+        id: create_issue_body
+        run: |
+          embedding_dimension=${{ github.event.inputs.embedding_dimension }}
+          pooling_mode=${{ github.event.inputs.pooling_mode }}
+          
+          issue_body="Please approve or deny opensearch-py-ml model uploading:
+          
+          ========= Workflow Details ==========
+          - Workflow Name: ${{ github.workflow }}
+          - Workflow Run ID: ${{ github.run_id }}
+          - Workflow Initiator: @${{ github.actor }}
+          
+          ========= Model Information =========
+          - Model ID: ${{ github.event.inputs.model_id }}
+          - Model Version: ${{ github.event.inputs.model_version }}
+          - Tracing Format: ${{ github.event.inputs.tracing_format }}
+          - Embedding Dimension: ${embedding_dimension:-Default}
+          - Pooling Mode: ${pooling_mode:-Default}
+          
+          ===== Dry Run of Model Uploading =====
+          ${{ needs.model-auto-tracing.outputs.dryrun_output }}"
+
+          echo "issue_body<<EOF" >> $GITHUB_OUTPUT
+          echo "${issue_body@E}" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+          echo "${issue_body@E}"
+      - uses: trstringer/manual-approval@v1
+        with:
+          secret: ${{ github.TOKEN }}
+          approvers: ${{ steps.get_approvers.outputs.approvers }}
+          minimum-approvals: 1
+          issue-title: "Upload Model to OpenSearch Model Hub (${{ github.event.inputs.model_id }})"
+          issue-body: ${{ steps.create_issue_body.outputs.issue_body }}
+          exclude-workflow-initiator-as-approver: false
+      
+  # Step 5: Download the artifacts & Upload it to the S3 bucket
+  model-uploading:
+    needs: manual-approval
+    runs-on: 'ubuntu-latest'
+    permissions:
+      id-token: write
+      contents: read
+    concurrency: ${{ github.workflow }}-concurrency
+    steps:
+      - name: Download Artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: upload
+          path: ./upload/
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: ${{ secrets.MODEL_UPLOADER_AWS_REGION }}
+          role-to-assume: ${{ secrets.MODEL_UPLOADER_ROLE }}
+          role-session-name: model-uploading
+      - name: Copy Files to the Bucket
+        id: copying_to_bucket
+        run: |
+          aws s3 sync ./upload/ s3://opensearch-exp/ml-models/huggingface/sentence-transformers/
```

Comment:
if we can keep this value in a variable that way we could reuse the same value.

Reply:
Will look into that. From what I know right now is that the variable in each `job` (i.e., model autotracing / manual approval / ...) is independent.